### ISSUE TITLE
Morphing fails for users with special characters

### ISSUE THREAD

**Issue by milescalabresi** [learning-unlimited/ESP-Website#2139]
When an admin tries to morph into a user with one or more special characters in their name, the server throws a "whoops." (The specific error is ```UnicodeEncodeError with Exception Value:	
'ascii' codec can't encode character```....) However, if you then try to visit a page that requires admin access, you must re-enter your admin credentials as if the morph were successful, but there is no "unmorph" tab as there usually would be, so the Unicode error causes part, but not all, of the morph process to fail.

**Comment by benjaminjkraft** [comment]
@milescalabresi do you have a traceback on that error?

**Comment by milescalabresi** [comment]
Yes, sorry. Here:
````
Traceback:
File "/home/vagrant/venv/local/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
  223.                 response = middleware_method(request, response)
File "/home/vagrant/devsite/esp/esp/middleware/espauthmiddleware.py" in process_response
  127.                 if request.COOKIES.get(key, "") != str(value if value else ""):

Exception Type: UnicodeEncodeError at /myesp/morph
Exception Value: 'ascii' codec can't encode character u'\xe1' in position ???: ordinal not in range(128)
````
I `???`'d out the number so as not to give away any personal data for the student I tested on. (There are several with the character in question in their names.)
It looks like `str()` isn't loving the special character, but I'm not sure where one has to go to tell Django to treat it as Unicode and not ASCII.

**Comment by benjaminjkraft** [comment]
Interesting.  Can you get it to print the key right above that line?  We seem to be attempting to encode the first/last name above (see the definition of `new_values`), so I'm wondering if the problem is in some other field that we aren't encoding, perhaps username, or maybe `retTitle` which is the return-to title, perhaps including the user's name from the user search page (and perhaps corroborated by that tab not showing).  If so, you can just do `urllib.quote(whatever.encode(encoding))` like we do for first/last name.

**Comment by milescalabresi** [comment]
You're right about `new_values`. The return-to title is `'cur_retTitle': u'User Search for F\xe1ke Student'` where the name is Fáke Student. They key and value are:
`value u'User Search for F\xe1ake Student`
`key 'cur_retTitle'`

Something else that seems a little strange is that the string that throws the error includes "for" (presumably from `User Search for`) instead of the first and last name.
`The string that could not be encoded/decoded was: `**`for Fáke`** instead of **`Fáke Student`**, as I would have guessed, but I don't know if that's contributing or symptomatic or actually correct.

I don't quite understand your suggestion. Where do you think this encoding line needs to be added?

**Comment by benjaminjkraft** [comment]
It may just show some snippet of the string.  In any case, it sounds like you want to do the quoting/encoding I mentioned above for `ret_title`.

**Comment by milescalabresi** [comment]
Fixed by #2148
